docs/developer-guide/extensions/proxy-extensions.md

```diff
@@ -0,0 +1,246 @@
+# Proxy Extensions
+*Current Status: [Alpha][1] (Since v2.7.0)*
+
+## Overview
+
+With UI extensions it is possible to enhance Argo CD web interface to
+provide valuable data to the user. However the data is restricted to
+the resources that belongs to the Application. With proxy extensions
+it is also possible to add additional functionality that have access
+to data provided by backend services. In this case Argo CD API server
+acts as a reverse-proxy authenticating and authorizing incoming
+requests before forwarding to the backend service.
+
+## Configuration
+
+As proxy extension is in [Alpha][1] phase, the feature is disabled by
+default. To enable it, it is necessary to configure the feature flag
+in Argo CD command parameters. The easiest way to to properly enable
+this feature flag is by adding the `server.enable.proxy.extension` key
+in the existing `argocd-cmd-params-cm`. For example:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: argocd-cmd-params-cm
+  namespace: argocd
+data:
+  server.enable.proxy.extension: "true"
+```
+
+Once the proxy extension is enabled, it can be configured in the main
+Argo CD configmap ([argocd-cm][2]).
+
+The example below demonstrate all possible configurations available
+for proxy extensions:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: argocd-cm
+  namespace: argocd
+data:
+  extension.config: |
+    extensions:
+    - name: httpbin
+      backend:
+        connectionTimeout: 2s
+        keepAlive: 15s
+        idleConnectionTimeout: 60s
+        maxIdleConnections: 30
+        services:
+        - url: http://httpbin.org
+          cluster: https://some-cluster
```

Comment:
```suggestion
          cluster: 
            # You have to specify at least one of these. We will check against whichever field is specified in the app spec.
            server: https://some-cluster
            name: some-cluster
```

Reply:
explained differently in the proper section below in the documentation.